doc/source/management/data_at_rest_encryption.rst

```diff
@@ -11,244 +11,470 @@ This feature is considered **BETA** quality.
-<https://dev.mysql.com/doc/refman/5.7/en/replication-options-binary-log.html#sysvar_master_verify_checksum>`_
-and `binlog_checksum
-<https://dev.mysql.com/doc/refman/5.7/en/replication-options-binary-log.html#sysvar_binlog_checksum>`_
-variables to be turned ``ON``.
+* |PXC| DONOR node invokes |xtrabackup| in backup stage providing on-the-fly
+  generated transition-key.
+* |xtrabackup| then re-encrypts the encrypted objects using this transition-key
+  there-by removing dependency on the existing ``keyring_plugin``.
+* Encrypted objects are then streamed to JOINER.
+* On JOINER side, |xtrabackup| will execute prepare/move-back stage using the
+  same transition-key (transition-key is streamed by the SST protocol). Since
+  we are sending over transition-key connection needs to be secured.
+* |xtrabackup| then re-encrypts the objects to generate new master-key that is
+  then stored in keyring plugin as configured on JOINER side. 
 
-While replicating, master sends the stream of decrypted binary log events to a
-slave (SSL connections can be set up to encrypt them in transport). That said,
-masters and slaves use separate keyring storages and are free to use differing
-keyring plugins.
+Using xtrabackup 2.4.11+
+------------------------
 
-Dumping of encrypted binary logs involves decryption, and can be done using
-``mysqlbinlog`` with ``--read-from-remote-server`` option.
+Starting from 2.4.11, |xtrabackup| is shipping its own version of 
+``keyring_file`` and ``keyring_vault`` plugin. It can find the plugin at
+default location but if user has installed |xtrabackup| in non-standard way
+(e.g. using binary) then |xtrabackup| plugin directory should be specified
+explicitly. 
 
-.. note:: Taking into account that ``--read-from-remote-server`` option  is only
-   relevant to binary logs, encrypted relay logs can not be dumped/decrypted
-   in this way.
+This limitation has influence on the xtrabackup-based SST. User should specify
+``xtrabackup-plugin-dir`` in the ``my.cnf`` configuration file under 
+``[xtrabackup]`` section to help pass proper plugin directory.
 
-System Variables
-----------------
+.. code-block:: text
 
-.. variable:: encrypt_binlog
+   [xtrabackup]
+   xtrabackup-plugin-dir=<PATH>/percona-xtrabackup-2.4.11-Linux-x86_64/lib/plugin/
 
-  :version 5.7.20-19: Implemented
-  :cli: ``--encrypt-binlog``
-  :dyn: No
-  :scope: Global
-  :vartype: Boolean
-  :default: ``OFF``
+Above transition logic is transparent to the end user and is backward
+compatible. DONOR node still can donate to new JOINER node, and JOINER
+knows that it should expect a ``keyring_file`` and not ``transition-key``.
 
-The variable turns on binary and relay logs encryption.
+.. note:: |PXC| 5.7.22 is now dependent on |xtrabackup| 2.4.12 in order to
+   fully support vault plugin functionality.
 
-.. _keyring_vault_plugin:
 
-Keyring Vault plugin
-====================
+Configuring PXC to use keyring_vault plugin
+===========================================
```

Comment:
subheading like keyring_file is missing here

Configuring PXC to use keyring_file plugin
          keyring_file